src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/NewAzureEventHubNameSpaceKey.cs

```diff
@@ -0,0 +1,69 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.EventHub.Models;
+using Microsoft.Azure.Commands.EventHub.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+    /// <summary>
+    /// 'New-AzureRmEventHubNamespaceKey' Cmdlet creates a new specified (PrimaryKey / SecondaryKey) key for the given EventHub Namespace Authorization Rule
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, EventHubNamespaceKeyVerb, SupportsShouldProcess = true), OutputType(typeof(ListKeysAttributes))]
+    public class NewAzureRmEventHubNamespaceKey : AzureEventHubsCmdletBase
+    {
+        
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroup { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NamespaceName { get; set; }
+        
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "Authorization Rule Name.")]        
+        [ValidateNotNullOrEmpty]
+        public string AuthorizationRuleName { get; set; }        
+
+        [Parameter(Mandatory = true,
+            Position = 3,
+            ParameterSetName = RegenerateKeySetName,
```

Comment:
If there are not multiple parameter sets, then remove the ParameterSetName from this attribute.  If there are multiople parameter sets, then you need to indicate the parameter set for each parameter and add the 'DefaultParameterSetName' setting to the Cmdlet attribute